src/scene/materials/standard-material.js

```diff
@@ -719,19 +729,21 @@ class StandardMaterial extends Material {
         else
             this.shaderOptBuilder.updateRef(options, scene, this, objDefs, staticLightList, pass, sortedLights);
 
+        // execute user callback to modify the options
         if (this.onUpdateShader) {
             options = this.onUpdateShader(options);
         }
 
         const library = device.getProgramLibrary();
-        this.shader = library.getProgram('standard', options);
+        const shader = library.getProgram('standard', options);
 
         if (!objDefs) {
             this.clearVariants();
-            this.variants[0] = this.shader;
+            this.variants[0] = shader;
```

Comment:
any idea what the purpose of this `!objDefs` block is?

Reply:
as far as I can tell, it has no purpose / logic. It was getting triggered by lightmapper, but even that line served no purpose. So I've removed it all.